syncer/syncer.go

```diff
@@ -716,6 +729,38 @@ func (s *Syncer) addCount(isFinished bool, queueBucket string, tp opType, n int6
 	}
 }
 
+func (s *Syncer) jobsIsEmpty() bool {
+	for idx := range s.jobs {
+		if len(s.jobs[idx]) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// updateReplicationLag calc syncer replication lag by job, it called after every batch dml job / one skip job / one ddl
+// job is flushed to target db.
+func (s *Syncer) updateReplicationLag(job *job, queueBucketName string) {
+	if job != nil {
+		// NOTE workerlagmap already init all key(queueBucketName) before syncer running.
+		s.workerLagMap[queueBucketName].Store(time.Now().Unix() + s.tsOffset.Load() - int64(job.eventHeader.Timestamp))
```

Comment:
we can check skip/ddl because we use max lag to udpdate metric, if there is no skip/ddl in bucket, the lag will be `0` ,any dml lag greater than 0 will be used

Reply:
I thought about it again,  if there is a very large ddl/skip lag in lag map,  the lag metrics will always update to the large one, this is obviously wrong.

Reply:
fixed now, skip/ddl job's log  is handled separately